File: src/main/java/me/kuku/simbot/entity/StepEntity.java
package me.kuku.simbot.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "step")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class StepEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	@OneToOne
	@JoinColumn(name = "qq")
	private QqEntity qqEntity;

	private String leXinPhone;
	private String leXinPassword;
	@Column(length = 5000)
	private String leXinCookie;
	private String leXinUserid;
	@Column(length = 5000)
	private String leXinAccessToken;

	private String miPhone;
	private String miPassword;
	@Column(length = 1000)
	private String miLoginToken;

	private Integer step;

	public StepEntity(QqEntity qqEntity){
		this.qqEntity = qqEntity;
		this.step = -1;
	}

	public StepEntity(String leXinPhone, String leXinPassword, String leXinCookie, String leXinUserid, String leXinAccessToken){
		this.leXinPhone = leXinPhone;
		this.leXinPassword = leXinPassword;
		this.leXinCookie = leXinCookie;
		this.leXinUserid = leXinUserid;
		this.leXinAccessToken = leXinAccessToken;
		step = -1;
	}

	public StepEntity(String miPhone, String miPassword, String miLoginToken){
		this.miPhone = miPhone;
		this.miPassword = miPassword;
		this.miLoginToken = miLoginToken;
		this.step = -1;
	}
}

File: src/main/kotlin/me/kuku/simbot/entity/StepEntity.kt
package me.kuku.simbot.entity

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import javax.annotation.Resource
import javax.persistence.*

@Entity
@Table(name = "step")
data class StepEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,
    @OneToOne
    @JoinColumn(name = "qq")
    var qqEntity: QqEntity? = null,

    var leXinPhone: String = "",
    var leXinPassword: String = "",
    @Column(length = 5000)
    var leXinCookie: String = "",
    var leXinUserid: String = "",
    @Column(length = 5000)
    var leXinAccessToken: String = "",

    var miPhone: String = "",
    var miPassword: String = "",
    var miLoginToken: String = "",

    var step: Int = -1
){
    companion object{
        fun getInstance(qqEntity: QqEntity): StepEntity{
            return StepEntity(qqEntity = qqEntity)
        }
        fun getInstance(leXinPhone: String, leXinPassword: String, leXinCookie: String, leXinUserid: String, leXinAccessToken: String): StepEntity{
            return StepEntity(leXinPhone = leXinPhone, leXinPassword = leXinPassword, leXinCookie = leXinCookie,
                leXinUserid = leXinUserid, leXinAccessToken = leXinAccessToken)
        }
        fun getInstance(miPhone: String, miPassword: String, miLoginToken: String): StepEntity{
            return StepEntity(miPhone = miPhone, miPassword = miPassword, miLoginToken = miLoginToken)
        }
    }
}


interface StepRepository: JpaRepository<StepEntity, Int>{
    fun findByQqEntity(qqEntity: QqEntity): StepEntity?
}

interface StepService{
    fun findByQqEntity(qqEntity: QqEntity): StepEntity?
    fun save(stepEntity: StepEntity): StepEntity
}

@Service
class StepServiceImpl: StepService{

    @Resource
    private lateinit var stepRepository: StepRepository

    override fun findByQqEntity(qqEntity: QqEntity): StepEntity? {
        return stepRepository.findByQqEntity(qqEntity)
    }

    override fun save(stepEntity: StepEntity): StepEntity {
        return stepRepository.save(stepEntity)
    }
}